tools/VscPreviewExtension/src/childProcessHost.ts

```diff
@@ -74,24 +65,62 @@ export class ChildProcessHost {
         let docContent = doc.getText();
         let fileName = doc.fileName;
         let rootPath = workspace.rootPath;
-        let filePath;
+        let relativePath;
         if (!rootPath || !fileName.includes(rootPath)) {
-            let indexOfFilename = fileName.lastIndexOf("\\");
-            rootPath = fileName.substr(0, indexOfFilename);
-            filePath = fileName.substring(indexOfFilename + 1);
+            let indexOfFileName = fileName.lastIndexOf("\\");
+            rootPath = fileName.substr(0, indexOfFileName);
+            relativePath = fileName.substring(indexOfFileName + 1);
         } else {
             let rootPathLength = rootPath.length;
-            filePath = fileName.substr(rootPathLength + 1, fileName.length - rootPathLength);
+            relativePath = fileName.substr(rootPathLength + 1, fileName.length - rootPathLength);
         }
-        if (doc.languageId === ConstVariable.languageId) {
-            let numOfRow = doc.lineCount;
-            this.writeToStdin(rootPath, filePath, numOfRow, docContent);
+        if (doc.languageId === "markdown") {
+            this.sendHttpRequestCoreAsync(rootPath, relativePath, docContent);
```

Comment:
That's right, when calling an async function without `await`, we can't catch the exception which threw by the function.

---
In reply to: [111743924](https://github.com/dotnet/docfx/pull/1547#discussion_r111743924) [](ancestors = 111743924)